### PR TITLE
Update CLI

### PR DESCRIPTION
**Pull request by bogdanPricope**
Update commands 'show', 'help', 'sysctl'.
Add static arp.

### CODE REVIEW

**Comment by matiaselo** [comment]
This is really useful commit for quick testing. Before I can merge the patch you should fill [the OFP Contribution Agreement](https://openfastpath.org/wp-content/uploads/2019/03/OpenFastPath_Contribution-License-Agreement.pdf) and send it to me. You can simply leave the Corporation part empty.

There are still quite a few checkpatch warnings which should be fixed. Some of the warnings can be ignored due to following the particular file's coding style.

File: src/cli/ofp_cli.c [hunk 1]
```diff
@@ -186,6 +186,35 @@ static int ip4net_ok(char *val)
...
 }
 
+static int mac_ok(char *val)
+{
+	char *tk_start, *tk_end, *val_end, *ch;
+	int tk_cnt = 0;
+
+	for(tk_cnt = 0, tk_start = val, val_end = val + strlen(val);
+		tk_start != val_end;
+		tk_cnt++, tk_start = (*tk_end == '\0')? tk_end: tk_end + 1)
```
Comment (matiaselo) [inline src/cli/ofp_cli.c]: This for loop is quite hard to understand and could probably be simplified a bit.

File: src/cli/ofp_cli_arp.c [hunk 2]
```diff
@@ -47,13 +47,62 @@ void f_arp_cleanup(struct cli_conn *conn, const char *s)
...
 	(void)s;
 	ofp_sendf(conn->fd,
 		"Show arp table:\r\n"
 		"  arp\r\n\r\n");
 
+	ofp_sendf(conn->fd,
+		"Add arp entry:\r\n"
+		"  arp set IP4ADDR HWADDR dev DEV\r\n"
+		"    IP4ADDR: ip address in a.b.c.d format\r\n"
+		"    HWADDR: mac address\r\n"
```
Comment (matiaselo) [inline src/cli/ofp_cli_arp.c]: For uniformity you could add the address format also for HWADDR or remove the format from IP4ADDR (the format is already show in the example).

**Comment by sovu** [comment]
@matiaselo OFP Contribution Agreement for Bogdan was sent and you can find it with the other CLAs.

**Comment by bogdanPricope** [comment]
I'll have a look tomorrow. I'm learning French and taking interviews... not much time for this.

**Comment by matiaselo** [comment]
> @matiaselo OFP Contribution Agreement for Bogdan was sent and you can find it with the other CLAs.

OK

File: include/ofpi_cli.h [hunk 3]
```diff
@@ -96,6 +96,8 @@ void f_address_add(struct cli_conn *conn, const char *s);
 void f_address_del(struct cli_conn *conn, const char *s);
 void f_address_help(struct cli_conn *conn, const char *s);
 
+
```
Comment (matiaselo) [inline include/ofpi_cli.h]: Extra newline

File: src/cli/ofp_cli.c [hunk 4]
```diff
@@ -186,6 +186,41 @@ static int ip4net_ok(char *val)
...
+	while (tk_start != val_end) {
+		tk_end = strchr(tk_start, ':');
+		if (tk_end == NULL)
+			tk_end = val_end;
+
+		for (pch = tk_start; pch != tk_end; pch++)
+			if (!((*pch >= '0' && *pch <= '9') ||
+				(*pch >= 'a' && *pch <= 'f') ||
+				(*pch >= 'A' && *pch <= 'F')))
+				return 0;
```
Comment (matiaselo) [inline src/cli/ofp_cli.c]: Alignment should match open parenthesis

File: src/cli/ofp_cli.c [hunk 5]
```diff
@@ -186,6 +186,41 @@ static int ip4net_ok(char *val)
...
+
+	while (tk_start != val_end) {
+		tk_end = strchr(tk_start, ':');
+		if (tk_end == NULL)
+			tk_end = val_end;
+
+		for (pch = tk_start; pch != tk_end; pch++)
+			if (!((*pch >= '0' && *pch <= '9') ||
+				(*pch >= 'a' && *pch <= 'f') ||
+				(*pch >= 'A' && *pch <= 'F')))
```
Comment (matiaselo) [inline src/cli/ofp_cli.c]: Alignment should match open parenthesis

File: src/cli/ofp_cli.c [hunk 6]
```diff
@@ -186,6 +186,41 @@ static int ip4net_ok(char *val)
...
+				return 0;
+
+		if ((tk_end - tk_start) != 2 &&
+			(tk_end - tk_start) != 1)
```
Comment (matiaselo) [inline src/cli/ofp_cli.c]: Alignment should match open parenthesis

File: src/cli/ofp_cli_arp.c [hunk 7]
```diff
@@ -47,13 +47,66 @@ void f_arp_cleanup(struct cli_conn *conn, const char *s)
...
+{
+	int a, b, c, d, e, f, g, h, i, j;
+	char dev[16];
+	int port, vlan;
+	struct ofp_ifnet *itf;
+	uint32_t ipv4_addr;
+	uint8_t mac[OFP_ETHER_ADDR_LEN];
+
+	if (sscanf(s, " %d.%d.%d.%d %x:%x:%x:%x:%x:%x %s",
+		&a, &b, &c, &d, &e, &f, &g, &h, &i, &j, dev) != 11)
```
Comment (matiaselo) [inline src/cli/ofp_cli_arp.c]: Alignment should match open parenthesis

File: src/cli/ofp_cli_arp.c [hunk 8]
```diff
@@ -47,13 +47,66 @@ void f_arp_cleanup(struct cli_conn *conn, const char *s)
...
 		"  arp\r\n\r\n");
 
+	ofp_sendf(conn->fd,
+		"Add arp entry:\r\n"
+		"  arp set IP4ADDR HWADDR dev DEV\r\n"
+		"    IP4ADDR: ip address in a.b.c.d format\r\n"
+		"    HWADDR: mac address in a:b:c:d:e:f format\r\n"
+		"    DEV: ethernet interface name\r\n"
+		"  Example:\r\n"
+		"    arp set 192.168.0.20 90:1b:0e:9a:cf:fc dev fp0\r\n\r\n");
```
Comment (matiaselo) [inline src/cli/ofp_cli_arp.c]: Alignment should match open parenthesis

**Comment by matiaselo** [comment]
Excluding the few remaining code alignment problems patch is looking good.

**Comment by bogdanPricope** [comment]
Ok, ok... I was missing the '.checkpatch.conf'.

**Comment by matiaselo** [comment]
Looking good. Please add the 'reviewed-by' tags to the commits and I'll merge them.

**Pull request merged**